### PR TITLE
Validate chat message input before send

### LINKED ISSUE #378: ChatViewModel validates outgoing message text before send
## Problem statement
Part of #213.

ChatViewModel must reject locally invalid outgoing message text before send submission.

## Proposed solution
Validate current input in ChatViewModel before creating TextMessage or calling SendMessageUseCase.

Reject:
- blank or whitespace-only text
- text longer than 2000 characters

On validation failure:
- update inputTextValidationError
- do not call SendMessageUseCase
- do not emit ClearInputText

## Acceptance criteria
- [ ] Blank messages are rejected locally with no send use case call.
- [ ] Whitespace-only messages are rejected locally with no send use case call.
- [ ] Messages longer than 2000 characters are rejected locally with no send use case call.
- [ ] Valid messages still submit normally.
- [ ] Tests cover validation and no-call behavior.

### PR DESCRIPTION
**Pull request by timurgilfanov**
Closes #378

## Summary
Validate outgoing chat message text locally in `ChatViewModel` before creating a `TextMessage` or invoking `SendMessageUseCase`.

The validation stays in `ChatViewModel` and reuses the existing `TextValidator(TextMessage.MAX_TEXT_LENGTH)` contract. No new validator, side effect, UI state field, or DI/API contract was added.

## Changes
- Reject blank and whitespace-only input with `TextValidationError.Empty` before entering the send flow.
- Reject input longer than `TextMessage.MAX_TEXT_LENGTH` with `TextValidationError.TooLong` before entering the send flow.
- Preserve valid send behavior, including one-time `ClearInputText` emission after successful sends.
- Extend chat ViewModel test fixtures to record sent messages and support repository failure results.

## Testing
- [x] Unit tests added or updated
- [ ] Manually tested
- [x] Edge cases considered

## Screenshots / Demo
Not applicable; behavior is covered at the ViewModel layer and no UI layout changed.

## Acceptance criteria
- [x] Blank messages are rejected locally with no send use case call.
- [x] Whitespace-only messages are rejected locally with no send use case call.
- [x] Messages longer than 2000 characters are rejected locally with no send use case call.
- [x] Valid messages still submit normally.
- [x] Tests cover validation and no-call behavior.

## Breaking Changes
None.

## Follow-ups
- [ ] None.

### CODE REVIEW

**Comment by timurgilfanov** [comment]
> 🧹 Nitpick comments (2)
> 
> > app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt (1)> `75-75`: _⚡ Quick win_
> > > **Inject `TextValidator` rather than instantiating it directly.**
> > > The ViewModel coding guidelines require "call injected validators on every field update." `TextValidator` is currently constructed inline, which means it cannot be replaced in unit tests or reconfigured without changing the ViewModel itself.
> > > 
> > > ♻️ Suggested change
> > > ```diff
> > >  class ChatViewModel `@AssistedInject` constructor(
> > >      `@Assisted`("chatId") chatIdUuid: UUID,
> > >      private val savedStateHandle: SavedStateHandle,
> > > +    private val textValidator: TextValidator,
> > >      private val sendMessageUseCase: SendMessageUseCase,
> > >      ...
> > >  ) : ViewModel() {
> > > -    private val textValidator = TextValidator(TextMessage.MAX_TEXT_LENGTH)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Provide the instance via a Hilt module (or bind it with `TextMessage.MAX_TEXT_LENGTH` as a qualifier/constant).
> > > 
> > > As per coding guidelines (`**/*ViewModel.kt`): "call injected validators on every field update and store the boolean result in state."

There is no current test or product need to vary text validation per VM, inline construction is more pragmatic.

File: app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt [hunk 1]
```diff
@@ -112,30 +112,44 @@ class ChatViewModel @AssistedInject constructor(
...
-                        if (result.data is TextMessage &&
-                            currentInputText == request.text
-                        ) {
-                            currentInputText = ""
-                            _effects.send(ChatSideEffect.ClearInputText)
-                        }
-                        _state.update {
-                            (it as? Ready)?.copy(isSending = false) ?: it
+        val inputTextValidationError = validateInputText(currentInputText)
+        if (inputTextValidationError != null) {
```
Comment (timurgilfanov) [inline app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt:116]: If text is empty will validator return error? Will sending button be disabled? Will error text be shown?

Comment (timurgilfanov) [reply app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt:116]: Yes.

At `sendMessage()`, `validateInputText(currentInputText)` uses `TextValidator.validate(text)`. For empty text, `text.isBlank()` is true, so it returns `TextValidationError.Empty`.

The send button will be disabled before that in normal UI flow because `onInputTextChanged("")` stores `inputTextValidationError = TextValidationError.Empty`, and `MessageInput` enables the button only when `textValidationError == null`.

No visible error text is shown for empty text. `MessageInput` maps it like this:

```kotlin
null, TextValidationError.Empty -> null
```

So empty input disables the send button without showing an error label. Only `TextValidationError.TooLong` shows visible error text.

**Review by timurgilfanov — COMMENTED** [review]
LGTM
